### ISSUE TITLE
`libnccl.so` is preloaded eagerly for CuPy wheel

### ISSUE THREAD

**Issue by tingyu66** [cupy/cupy#8354]
### Description

CuPy is designed to lazy-load CUDA libraries upon importing `cupy`, but this behavior changes when using CuPy wheel. In that case, a specific version of libnccl.so is loaded immediately:
```
root@1cc5aab-lcedt:/# LD_DEBUG=libs python -c "import cupy" 2>&1 | grep nccl.so
       480:	find library=libnccl.so.2.16.2 [0]; searching
       480:	  trying file=/pyenv/versions/3.9.19/lib/libnccl.so.2.16.2
       480:	  trying file=/lib/x86_64-linux-gnu/libnccl.so.2.16.2
       480:	calling init: /lib/x86_64-linux-gnu/libnccl.so.2.16.2
       480:	calling fini: /lib/x86_64-linux-gnu/libnccl.so.2.16.2 [0]
```
This causes issues when using other packages that requires a different version of NCCL. See relevant discussion [here](https://github.com/rapidsai/cugraph/issues/4465). 

CC: @leofang 

### To Reproduce

```bash
root@1cc5aab-lcedt:/# pip install cupy-cuda11x
root@1cc5aab-lcedt:/# LD_DEBUG=libs python -c "import cupy" 2>&1 | grep nccl.so
       480:	find library=libnccl.so.2.16.2 [0]; searching
       480:	  trying file=/pyenv/versions/3.9.19/lib/libnccl.so.2.16.2
       480:	  trying file=/lib/x86_64-linux-gnu/libnccl.so.2.16.2
       480:	calling init: /lib/x86_64-linux-gnu/libnccl.so.2.16.2
       480:	calling fini: /lib/x86_64-linux-gnu/libnccl.so.2.16.2 [0]

```

### Installation

Wheel (`pip install cupy-***`)

### Environment

```
OS                           : Linux-6.5.0-35-generic-x86_64-with-glibc2.31
Python Version               : 3.9.19
CuPy Version                 : 13.1.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.26.4
SciPy Version                : None
Cython Build Version         : 0.29.36
Cython Runtime Version       : None
CUDA Root                    : /usr/local/cuda
nvcc PATH                    : /usr/local/cuda/bin/nvcc
CUDA Build Version           : 11080
CUDA Driver Version          : 12050
CUDA Runtime Version         : 11080 (linked to CuPy) / 11080 (locally installed)
cuBLAS Version               : (available)
cuFFT Version                : 10900
cuRAND Version               : 10300
cuSOLVER Version             : (11, 4, 1)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 8)
Thrust Version               : 200200
CUB Build Version            : 200200
Jitify Build Version         : <unknown>
cuDNN Build Version          : (not loaded; try `import cupy.cuda.cudnn` first)
cuDNN Version                : (not loaded; try `import cupy.cuda.cudnn` first)
NCCL Build Version           : 21602
NCCL Runtime Version         : 21602
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA RTX A6000
Device 0 Compute Capability  : 86
Device 0 PCI Bus ID          : 0000:61:00.0

```

**Comment by kmaehashi** [comment]
Thanks for the feedback @tingyu66! I've sent a fix in #8360.

**Comment by kmaehashi** [comment]
We'll include the fix in v13.2.0 planned this week.
